### PR TITLE
Add target date sorting AB#50686

### PR DESCRIPTION
**Pull request by RobOatesHistoricEngland**
[AB#50686](https://hedev.visualstudio.com/76e71f30-b1ad-438a-a096-89ac98712f4e/_workitems/edit/50686)

### CODE REVIEW

**Review by csmith-he — DISMISSED** [review]
Looks good to me.  Running locally on my machine as expected.

File: arches_her/views/active_consultations.py [hunk 1]
```diff
@@ -76,26 +77,33 @@ def get(self, request):
...
+            else:
                 return self.get_paginated_data(grouped_tile_list, page_ct, page_num)
 
         return HttpResponseNotFound()
 
+    def remove_prn_prefix(self, order_param, data):
+        if order_param.startswith("Casework Officer"):
+            return re.sub(r"^\[\d+\] ", "", data)
+        return data
+
```
[the remark this answers is not in the document]
Comment (RobOatesHistoricEngland) [reply arches_her/views/active_consultations.py]: Removed

File: arches_her/views/active_consultations.py [hunk 2]
```diff
@@ -76,26 +77,33 @@ def get(self, request):
...
-                            key=lambda resource: resource[order_config[order_param][0]],
-                            reverse=order_config[order_param][1],
-                        )
-                    except KeyError as e:
-                        print("Error: ", e)
+            if order_param in list(order_config.keys()) and order_param is not None:
+                try:
+                    grouped_tile_list = sorted(
+                        grouped_tile_list,
+                        key=lambda resource: self.remove_prn_prefix(order_param, resource[order_config[order_param][0]]),
```
[the remark this answers is not in the document]
Comment (RobOatesHistoricEngland) [reply arches_her/views/active_consultations.py]: Removed

**Review by aj-he — APPROVED** [review]
Great! Approved

**Comment by aj-he** [comment]
AB#49511

**Pull request merged**